qubesadmin/tools/qvm_start_daemon.py

```diff
@@ -813,23 +937,36 @@ def main(args=None):
                 pid = int(pidfile.read().strip())
             os.kill(pid, signal.SIGHUP)
         except (FileNotFoundError, ValueError) as e:
-            parser.error('Cannot open pidfile {}: {}'.format(pidfile_path,
-                                                             str(e)))
+            parser.error(f'Cannot open pidfile {pidfile_path}: {str(e)}')
     else:
         loop = asyncio.get_event_loop()
         tasks = []
         for vm in args.domains:
             if vm.is_running():
-                tasks.append(asyncio.ensure_future(launcher.start_gui(
-                    vm, force_stubdom=args.force_stubdomain)))
-                tasks.append(asyncio.ensure_future(launcher.start_audio(
-                    vm)))
+                if "guivm" in enabled_services:
+                    tasks.append(
+                        asyncio.ensure_future(
+                            launcher.start_gui(
+                                vm,
+                                force_stubdom=args.force_stubdomain
+                            )
+                        )
+                    )
+                if "audiovm" in enabled_services:
+                    tasks.append(
+                        asyncio.ensure_future(
+                            launcher.start_audio(vm)
+                        )
+                    )
         if tasks:
             loop.run_until_complete(asyncio.wait(tasks))
         loop.stop()
         loop.run_forever()
         loop.close()
 
+    # Cleanup any started daemon
+    launcher.cleanup()
```

Comment:
This sound like a bad idea, `qvm-start-daemon` is responsible for starting those daemons, but restarting `qvm-start-daemon` (for any reason) shouldn't kill all the audio/gui daemons...